urla134d66615_fikr4n_itl_java_tgz-pJ8-example_ExampleJ8/src/org/arabeyes/itl/prayer/PrayerTimeCalc.java

```diff
@@ -0,0 +1,720 @@
+					if (tempPrayer[0] == 99) {
+						if (method.getExtremeLatitude() == ExtremeLatitude.HALF_INVALID)
+							tempPrayer[0] = portion
+							- (method.getFajrInv() / 60.0);
+						else
+							tempPrayer[0] = tempPrayer[1] - portion;
+						pt.fajr().setExtreme(true);
+					}
+					if (tempPrayer[5] == 99) {
+						if (method.getExtremeLatitude() == ExtremeLatitude.HALF_INVALID)
+							tempPrayer[5] = portion
+							+ (method.getIshaaInv() / 60.0);
+						else
+							tempPrayer[5] = tempPrayer[4] + portion;
+						pt.isha().setExtreme(true);
+					}
+				} else { /* for the always methods */
+					
+					if (method.getExtremeLatitude() == ExtremeLatitude.HALF_ALWAYS) {
+						tempPrayer[0] = portion - (method.getFajrInv() / 60.0);
+						tempPrayer[5] = portion + (method.getIshaaInv() / 60.0);
+					}
+					
+					else {
+						tempPrayer[0] = tempPrayer[1] - portion;
+						tempPrayer[5] = tempPrayer[4] + portion;
+					}
+					pt.fajr().setExtreme(true);
+					pt.isha().setExtreme(true);
+				}
+			} else if (ext == ExtremeLatitude.MIN_ALWAYS) {	
+				/* Do nothing here because this is implemented through fajrInv and
+				 * ishaaInv structure members */
+				tempPrayer[0] = tempPrayer[1];
+				tempPrayer[5] = tempPrayer[4];
+				pt.fajr().setExtreme(true);
+				pt.isha().setExtreme(true);
+			} else if (ext == ExtremeLatitude.MIN_INVALID) {				
+				if (tempPrayer[0] == 99) {
+					exinterval = (int) ((double) method.getFajrInv() / 60.0);
+					tempPrayer[0] = tempPrayer[1] - exinterval;
+					pt.fajr().setExtreme(true);
+				}
+				if (tempPrayer[5] == 99) {
+					exinterval = (int) ((double) method.getIshaaInv() / 60.0);
+					tempPrayer[5] = tempPrayer[4] + exinterval;
+					pt.isha().setExtreme(true);
+				}
+			} /* end switch */
+		} /* end extreme */
+		
+		/* Apply intervals if set */
+		if (method.getExtremeLatitude() != ExtremeLatitude.MIN_INVALID
+				&& method.getExtremeLatitude() != ExtremeLatitude.HALF_INVALID
+				&& method.getExtremeLatitude() != ExtremeLatitude.HALF_ALWAYS) {
+			if (method.getFajrInv() != 0)
+				tempPrayer[0] = tempPrayer[1] - (method.getFajrInv() / 60.0);
+			if (method.getIshaaInv() != 0)
+				tempPrayer[5] = tempPrayer[4] + (method.getIshaaInv() / 60.0);
+		}
+		
+		/* Final Step: Fill the Prayer array by doing decimal degree to
+		 * Prayer structure conversion*/
+		if (type == InternalTimeType.IMSAK /* || type == TimeType.NEXT_FAJR */) {
+			base6hm(tempPrayer[0], method, pt.fajr(), type);
+		} else {
+			PrayerTime[] pArray = pt.getPrayerTime();
+			InternalTimeType[] timeArray = { InternalTimeType.FAJR, InternalTimeType.SUNRISE,
+					InternalTimeType.ZUHR, InternalTimeType.ASR, InternalTimeType.MAGHRIB,
+					InternalTimeType.ISHA };
+			
+			for (i = 0; i < 6; i++) {
+				base6hm(tempPrayer[i], method, pArray[i], timeArray[i]);
+			}
+		}
+		
+	}
+
+	private void base6hm(double bs, Method method, PrayerTime pt, InternalTimeType type) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 20. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 549-556 into a separate method.